source/crud/tests/v2/updateWithPipelines.yml

```diff
@@ -0,0 +1,82 @@
+data:
+    - { _id: 1, x: 1, y: 1, t: {u: {v: 1}} }
+    - { _id: 2, x: 2, y: 1 }
+
+minServerVersion: '4.1.11'
+
+collection_name: &collection_name "test"
+database_name: &database_name "crud-tests"
+
+tests:
+  -
+    description: "UpdateOne using pipelines"
+    operations:
+      -
+        name: "updateOne"
+        arguments:
+            filter: { y: 1 }
+            update: [ { $replaceRoot: { newRoot: "$t" } }, { $addFields: { foo: 1 } } ]
+    expectations:
+      -
+        command_started_event:
+          command:
+            update: *collection_name
+            updates:
+              -
+                q: { y: 1 }
+                u: [ { $replaceRoot: { newRoot: "$t" } }, { $addFields: { foo: 1 } } ]
+          command_name: update
+          database_name: *database_name
+    outcome:
+      collection:
+        data:
+          - { _id: 1, x: 1, y: 1, u: {v: 1}, foo: 1 }
+          - { _id: 2, x: 2, y: 1 }
+  -
+    description: "UpdateMany using pipelines"
+    operations:
+      -
+        name: "updateMany"
+        arguments:
+            filter: {}
+            update: [ { $project: { x: 1 } }, { $addFields: { foo: 1 } } ]
+    expectations:
+      -
+        command_started_event:
+          command:
+            update: *collection_name
+            updates:
+              -
+                q: { }
+                u: [ { $project: { x: 1 } }, { $addFields: { foo: 1 } } ]
+          command_name: update
+          database_name: *database_name
+    outcome:
+      collection:
+        data:
+          - { _id: 1, x: 1, foo: 1 }
+          - { _id: 2, x: 2, foo: 1 }
+  -
+    description: "FindOneAndUpdate using pipelines"
+    operations:
+      -
+        name: "findOneAndUpdate"
+        arguments:
+            filter: { y: 1 }
+            update: [ { $project: { x: 1 } }, { $addFields: { foo: 1 } } ]
```

Comment:
Would it make sense for this to include a `result` assertion like other `findOneAndUpdate` tests?

Reply:
Added `result` assertion to the test.